src/new_hope/week1/day2.clj

```diff
@@ -0,0 +1,59 @@
+(ns new-hope.week1.day2
+  (:require [midje.sweet :refer :all]
+            [criterium.core :as crit]))
+
+
+;The	rest	function	will	return	all	the	items	of	a	sequence	except	the	first.
+(facts "rest func"
+       (rest  [10  20  30  40]) => [20 30 40])
+
+
+;Intro	to	Functions
+;Clojure	has	many	different	ways	to	create	functions.
+
+(facts "func 1"
+       8 =>  ((fn  add-five  [x]  (+  x  5))  3))
+(facts "func 2"
+       8 => ((fn  [x]  (+  x  5))  3))
+(facts "func 3"
+       8 => (#(+ %  5)  3))
+(facts "func 4"
+       8 =>((partial  +  5)  3))
+
+;Write	a	function	which	doubles	a	number.
+(defn num-double [x] (* x 2))
+(facts "func 5"
+       (num-double  2)  4)
+(facts "func 6"
+       (num-double  3)  6)
+(facts "func 7"
+       (num-double  11)  22)
+(facts "func 8"
+       (num-double  7)  14)
+
+;Write	a	function	which	returns	a	personalized	greeting.
+(defn hello [x] str "Hello, " x)
```

Comment:
This actually does not work, please fix the tests first.
`(hello "Dave")` returns just `"Dave"`